src/authx/auth.py

```diff
@@ -598,3 +599,35 @@ def get_service_store_secret(service, key=None, vault_url=VAULT_URL, role_id=Non
         result = response.json()["data"]
         return result, 200
     return response.text, response.status_code
+
+def create_service_token(vault_url=VAULT_URL):
+    if SERVICE_NAME is None:
+        raise CandigAuthError("No SERVICE_NAME specified. Was this called from a CanDIG docker container?")
+    # create the random token:
+    token = uuid.uuid1()
+    try:
+        response, status_code = set_service_store_secret(SERVICE_NAME, key=f"token/{token}", value={"token": token})
+        if status_code != 200:
+            raise CandigAuthError(f"Could not create_service_token from {SERVICE_NAME}: {response}")
+    except Exception as e:
+        raise CandigAuthError(f"Could not create_service_token from {SERVICE_NAME}: {str(e)}")
+    return str(token)
+
+def verify_service_token(service=None, token=None):
+    if service is None:
+        return False
+    if token is None:
+        return False
+    body = {
+        "input": {
+            "service": service,
+            "token": token
+        }
+    }
+
+    response = requests.post(
+        OPA_URL + "/v1/data/service/verified",
```

Comment:
The value of `OPA_URL` is `http://candig.docker.internal:5080/policy`, which requires a valid auth token to access. I think this should be the internal url (`OPA_PRIVATE_URL` from the .env?)

Reply:
Oh! I think maybe can you try logging in and not running env.sh? I think that one is overriding the other value of OPA_URL from the docker-compose.

Reply:
Running without env.sh causes it to error out with some other errrors:
```
========================================================================================= short test summary info ==========================================================================================
FAILED test_auth.py::test_site_admin - AssertionError: assert False
FAILED test_auth.py::test_user_email - AssertionError: assert None == 'None@test.ca'
FAILED test_auth.py::test_get_opa_datasets - requests.exceptions.HTTPError: 401 Client Error: Unauthorized for url: http://opa:8181/v1/data/permissions/datasets
FAILED test_auth.py::test_put_aws_credential - assert 400 == 200
================================================================================= 4 failed, 7 passed, 9 warnings in 0.60s ==================================================================================
```

Running env.sh with the OPA_URL does work though. We should think of a way to make this testable without manually editing env.sh though... hmmm

Reply:
I am loath to put a bunch of stuff into the docker-compose just for manual testing, but you could put them in

You could manually edit the container's copy of env.sh...I think that it looks like you can just comment out OPA_URL so that it doesn't override that one?

Reply:
That's fair haha, I have no idea how to do this without being clunky. Let me look around a bit more, but otherwise this looks good :heavy_check_mark:

Reply:
Is this resolved enough to approve?